### DISCUSSION TITLE
argument of daily_file_sink

### DISCUSSION THREAD

**Discussion by 330305020** [gabime/spdlog#2840] [Q&A]
For daily_file_sink , it mentioned that 
/*
 * Rotating file sink based on date.
 * If truncate != false , the created file will be truncated.
 * If max_files > 0, retain only the last max_files and delete previous.
 */
I have two questions:
1. if truncate is set to true, based on which criteria the created file will be truncated? file size?
2. max_files: if one day the process is stopped, file of that day is not created, that means one file is missing, then files before process stop will not be deleted.
![image](https://github.com/gabime/spdlog/assets/44339104/794641c8-f140-49bc-8207-043df7a8043b)

**Comment by tt4g** [comment]
> 1. if truncate is set to true, based on which criteria the created file will be truncated? file size?

This is the same behavior as if the `O_TRUNC` flag were specified for `open()` for log files.

https://github.com/gabime/spdlog/blob/2ee8bac78e6525a8ad9a9196e65d502ce390d83a/include/spdlog/details/file_helper-inl.h#L32-L60

> 2. max_files: if one day the process is stopped, file of that day is not created, that means one file is missing, then files before process stop will not be deleted.

This is a specification.

You are only considering a pattern where there are no logfiles for one day old, but what do you think would happen if there were only 10 logfiles when `max_file = 20`?
Since no file older than 11 days exists, `path_exists?(filename)` will always return `false`, resulting in an infinite loop.
